libblkid/src/topology/ioctl.c

```diff
@@ -48,18 +54,20 @@ static int probe_ioctl_tp(blkid_probe pr,
 		const struct topology_val *val = &topology_vals[i];
 		int rc = 1;
 		union {
-			unsigned long ul;
 			int i;
 			uint64_t u64;
 		} data;
 
```

Comment:
Initializing the union to zero would be more robust than calling the ioctl with some random data.

Reply:
It's purely an output value.

Reply:
I know, but one day, any static analyzer will be unhappy with it.

Reply:
(Sorry, for all the pedantic notes, but it would be nice to have it readable and clean for future readers.)

Reply:
Not sure how it can be unhappy with it.
But it doesn't hurt, so done.